### PR TITLE
Fixing a bug with --disable-security-check flag

### PR DESCRIPTION
**Pull request by agrawalravi90**
#### Describe Bug or Feature

We were disabling security always, this fixes it

### CODE REVIEW

File: m4/security_check.m4 [hunk 1]
```diff
@@ -39,18 +39,18 @@
...
   )
-  AS_IF([test "x$disable_security_check" != "xyes"],
-    AC_MSG_RESULT([yes])
-    AC_DEFINE([NO_SECURITY_CHECK], [1], [Define as 1 to disable security check, 0 to enable]),
+  AS_IF([test "x$enable_security" != "xno"],
     AC_MSG_RESULT([no])
-    AC_DEFINE([NO_SECURITY_CHECK], [0], [Define as 1 to disable security check, 0 to enable])
+    AS_ECHO("Security checks will be performed"),
+    AC_MSG_RESULT([yes])
+    AC_DEFINE([NO_SECURITY_CHECK], [1], [Define as 1 to disable security check, 1 to enable])
```
Comment (vstumpf) [inline m4/security_check.m4:54]: I don't understand this comment, 1 to disable, 1 to enable.
Should this be something like
`AC_DEFINE([NO_SECURITY_CHECK], [], [Defined when skipping security checks])`

**Review by vstumpf — CHANGES_REQUESTED** [review]

**Review by arungrover — APPROVED** [review]
LGTM

**Review by vstumpf — APPROVED** [review]
LGTM

**Comment by vstumpf** [comment]
Here's some test logs:
Default:
```
[vstumpf@shecil pbspro]$ ./configure | grep security
checking whether to disable security check... no
[vstumpf@shecil pbspro]$ cat src/include/pbs_config.h | grep SECURITY
/* #undef NO_SECURITY_CHECK */
/* #undef PBS_SECURITY */
```

enable security checks
```
[vstumpf@shecil pbspro]$ ./configure --enable-security | grep security
checking whether to disable security check... no
[vstumpf@shecil pbspro]$ cat src/include/pbs_config.h | grep SECURITY
/* #undef NO_SECURITY_CHECK */
/* #undef PBS_SECURITY */
[vstumpf@shecil pbspro]$ 
```

disable security checks
```
[vstumpf@shecil pbspro]$ ./configure --disable-security | grep security
checking whether to disable security check... yes
[vstumpf@shecil pbspro]$ cat src/include/pbs_config.h | grep SECURITY
#define NO_SECURITY_CHECK /**/
/* #undef PBS_SECURITY */
```

**Comment by agrawalravi90** [comment]
Thanks for the merge @vstumpf !

**Pull request merged**